### PR TITLE
fall back to zram if no swap partition or file was found

### PR DESCRIPTION
**Pull request by dimkr**

### CODE REVIEW

File: woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit [hunk 1]
```diff
@@ -60,8 +60,8 @@ status_func() {
 
 create_zram_swap() {
   modprobe zram 2>/dev/null
-  # compression ratio should be in the range of 2:1 to 3:1; 1 GB will bring 32-bit machines with 2 GB of RAM to the limit of 4 GB
-  echo 1G > /sys/block/zram0/disksize || return 1
+  RAM=`awk '/MemTotal/{print $2}' /proc/meminfo`
```
Comment (01micko) [inline woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit:63]: Better idea!

File: woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit [hunk 2]
```diff
@@ -58,6 +58,14 @@ status_func() {
  return $1 #return with same status as entry param.
 }
 
+create_zram_swap() {
```
Comment (01micko) [inline woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit:61]: perhaps we `return 1` if we run an older kernel not supporting zram here?

File: woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit [hunk 3]
```diff
@@ -58,6 +58,14 @@ status_func() {
...
+  modprobe zram 2>/dev/null
```
Comment (01micko) [inline woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit:62]: woops, should be this line
`modprobe zram 2>/dev/null || return 1`

File: woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit (hunk 2 above)
Comment (dimkr) [reply woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit:61]: zram can be builtin. The `echo` will fail, if zram wasn't loaded and isn't builtin.

File: woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit (hunk 3 above)
Comment (dimkr) [reply woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit:62]: See my previous comment

Comment (01micko) [reply woof-code/rootfs-skeleton/etc/rc.d/rc.sysinit:62]: Ok, I'll remove the `|| return 1` from 2066bed

**Comment by dimkr** [comment]
@01micko Was it your intention to close this PR without merging it? Personally, I want it merged (great for my 2 GB laptop), but I'm open to feedback and even objections.

**Comment by 01micko** [comment]
Not at all! I changed it and added credit to you in 2066bed Then you indicated that maybe zram could be builtin so I fixed it, so my bad. If you want, I can revert my commits and just pull the PR!

**Comment by dimkr** [comment]
> Not at all! I changed it and added credit to you in [2066bed](https://github.com/puppylinux-woof-CE/woof-CE/commit/2066bed3081f683b76272b51fa5cfbac25dd0b76) Then you indicated that maybe zram could be builtin so I fixed it, so my bad. If you want, I can revert my commits and just pull the PR!

Thanks, I got confused. I thought you pushed a commit to this PR, and not to testing, then closed the PR.

**Pull request closed without merging**